lib/index.js

```diff
@@ -80,6 +86,13 @@ Optimizely.prototype.track = function(track) {
   var revenue = track.revenue();
   if (revenue) payload.revenue = revenue *= 100;
   push(['trackEvent', track.event(), payload]);
+
+  var optimizelyClientInstance = window.optimizelyClientInstance;
+  if (optimizelyClientInstance && optimizelyClientInstance.track) {
+    var attributes = track.properties();
+    var userId = attributes.userId;
```

Comment:
@mikeng13 How are you testing this? Are you just using that gist on a html page? I have a feeling this might be due to the fact that you are adding all this code inside our snippet.

Can you leave the snippet as is and basically do this instead using our `.ready()` function?

```
<html>
  <head>
    <script type="text/javascript">
      // LOADS THE SEGMENT ANALYTICS LIB
      !function(){var analytics=window.analytics=window.analytics||[];if(!analytics.initialize)if(analytics.invoked)window.console&&console.error&&console.error("Segment snippet included twice.");else{analytics.invoked=!0;analytics.methods=["trackSubmit","trackClick","trackLink","trackForm","pageview","identify","reset","group","track","ready","alias","debug","page","once","off","on"];analytics.factory=function(t){return function(){var e=Array.prototype.slice.call(arguments);e.unshift(t);analytics.push(e);return analytics}};for(var t=0;t<analytics.methods.length;t++){var e=analytics.methods[t];analytics[e]=analytics.factory(e)}analytics.load=function(t){var e=document.createElement("script");e.type="text/javascript";e.async=!0;e.src=("https:"===document.location.protocol?"https://":"http://")+"cdn.segment.com/analytics.js/v1/"+t+"/analytics.min.js";var n=document.getElementsByTagName("script")[0];n.parentNode.insertBefore(e,n)};analytics.SNIPPET_VERSION="4.0.0";
      // INITIALIZE SEGMENT
      analytics.load("YOUR_KEY_HERE");
      analytics.page();
      }}();

      analytics.ready(function(){
         analytics.identify(userId);
      // INITIALIZE OPTIMIZELY
      var datafile = {}; // YOUR DATAFILE HERE
      var optimizelyClientInstance = window.optimizelyClient.createInstance({ datafile: datafile });
      
      // ACTIVATE EXPERIMENT TO RECORD AN IMPRESSION BEFORE WE START TRACKING EVENTS
      optimizelyClientInstance.activate('experiment_1', userId);
      
      // TRACK EVENT WITH SEGMENT
      analytics.track("page_loaded", { 'metadata': 'add your custom metadata, attributes, traits here as individual properties' });
      });
    </script>
  </head>
</html>
```

The `.ready()` callback function is executed after we finish initializing all the integrations so it's best to do any post init logic in there.